engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestEmMinMax.java

```diff
@@ -356,73 +358,103 @@ public void testThrowBehaviors() {
                 () -> bytes.updateBy(UpdateByOperation.EmMin(throwControl, 10)));
+        Assert.eqTrue(err.getCause().getClass() == TableDataException.class,
+                "err.getCause().getClass() == TableDataException.class");
 
 
-        assertThrows(TableDataException.class,
+        err = assertThrows(TableInitializationException.class,
                 () -> bytes.updateBy(UpdateByOperation.EmMin(throwControl, 10)));
+        Assert.eqTrue(err.getCause().getClass() == TableDataException.class,
+                "err.getCause().getClass() == TableDataException.class");
 
         TableDefaults shorts = testTable(RowSetFactory.flat(4).toTracking(),
                 shortCol("col", (short) 0, (short) 1, NULL_SHORT, (short) 3));
 
-        assertThrows(TableDataException.class,
+        err = assertThrows(TableInitializationException.class,
                 () -> shorts.updateBy(UpdateByOperation.EmMin(throwControl, 10)));
+        Assert.eqTrue(err.getCause().getClass() == TableDataException.class,
+                "err.getCause().getClass() == TableDataException.class");
 
         TableDefaults ints = testTable(RowSetFactory.flat(4).toTracking(),
                 intCol("col", 0, 1, NULL_INT, 3));
 
-        assertThrows(TableDataException.class,
+        err = assertThrows(TableInitializationException.class,
                 () -> ints.updateBy(UpdateByOperation.EmMin(throwControl, 10)));
+        Assert.eqTrue(err.getCause().getClass() == TableDataException.class,
+                "err.getCause().getClass() == TableDataException.class");
 
         TableDefaults longs = testTable(RowSetFactory.flat(4).toTracking(),
                 longCol("col", 0, 1, NULL_LONG, 3));
 
-        assertThrows(TableDataException.class,
+        err = assertThrows(TableInitializationException.class,
                 () -> longs.updateBy(UpdateByOperation.EmMin(throwControl, 10)));
+        Assert.eqTrue(err.getCause().getClass() == TableDataException.class,
+                "err.getCause().getClass() == TableDataException.class");
 
         TableDefaults floats = testTable(RowSetFactory.flat(4).toTracking(),
                 floatCol("col", 0, 1, NULL_FLOAT, Float.NaN));
 
-        assertThrows(TableDataException.class,
+        err = assertThrows(TableInitializationException.class,
                 () -> floats.updateBy(
                         UpdateByOperation.EmMin(OperationControl.builder().onNullValue(BadDataBehavior.THROW).build(),
-                                10)),
-                "Encountered null value during EMS processing");
-
-        assertThrows(TableDataException.class,
+                                10)));
+        err = err.getCause();
+        Assert.eqTrue(err.getClass() == TableDataException.class,
+                "err.getClass() == TableDataException.class");
+        Assert.eqTrue(err.getMessage().contains("Encountered null value during Exponential Moving output processing"),
```

Comment:
This doesn't match the previous message. I don't recall a change to the message in your original PR.

Reply:
Ah. It appears that the old code was not validating anything at all. The message was for the assertion error, not for validation. I don't love message validations, but at least I see why this is correct now.